runtime/src/accounts.rs

```diff
@@ -165,30 +142,34 @@ impl AccountStorageEntry {
         self.count.fetch_add(1, Ordering::Relaxed);
     }
 
-    fn remove_account(&self) {
+    fn remove_account(&self) -> bool {
         if self.count.fetch_sub(1, Ordering::Relaxed) == 1 {
             self.accounts.reset();
             self.set_status(AccountStorageStatus::StorageAvailable);
+            true
+        } else {
+            false
         }
     }
 }
 
-type AccountStorage = Vec<AccountStorageEntry>;
+type AccountStorage = Vec<Arc<AccountStorageEntry>>;
+type AccountStorageSlice = [Arc<AccountStorageEntry>];
 
 // This structure handles the load/store of the accounts
 #[derive(Default)]
 pub struct AccountsDB {
     /// Keeps tracks of index into AppendVec on a per fork basis
-    account_index: AccountIndex,
+    accounts_index: RwLock<AccountsIndex<AccountInfo>>,
```

Comment:
I"ve always hated the "Info" types in this module.  Can we change this to AcccountLocation, now?